[BUGFIX beta] Allow feature flags to be merged.

rwjblue opened the pull request:
Previously, having `EmberENV.FEATURES` specified as an empty object (the default for ember-cli), meant that development builds of Ember would not have ANY feature flags (even those that are supposed to be always enabled).

This changes that, by starting with the default features listing and only changing those that are specified.

Closes #10901.

stefanpenner commented on packages/ember-metal/lib/core.js:
```diff
@@ -94,11 +94,12 @@ if ('undefined' === typeof Ember.ENV.DISABLE_RANGE_API) {
-  Ember.FEATURES = DEFAULT_FEATURES; //jshint ignore:line
+if (Ember.ENV.FEATURES) {
+  for (var feature in Ember.ENV.FEATURES) {
+    Ember.FEATURES[feature] = Ember.ENV.FEATURES[feature];
```
likely need a hasOwnProperty check

rwjblue replied:
Makes sense. Will update. 

I'm also digging into why the nose tests are failing here (most likely because I am manually manipulating the feature flags there).

rwjblue committed:
[BUGFIX beta] Allow feature flags to be merged.

Previously, having `EmberENV.FEATURES` specified as an empty object (the
default for ember-cli), meant that development builds of Ember would not
have ANY feature flags (even those that are supposed to be always
enabled).

This changes that, by starting with the default features listing and
only changing those that are specified.

rwjblue committed:
Bump emberjs-build to ensure default features are available to runtime tree.
